src/js/common/schemaform/definitions/address.js

```diff
@@ -120,12 +124,22 @@ export function uiSchema(label = 'Address', useStreet3 = false) {
       validateAddress
     ],
     'ui:options': {
-      updateSchema: (formData, addressSchema, addressUiSchema, index, path) =>
-        addressChangeSelector({
+      updateSchema: (formData, addressSchema, addressUiSchema, index, path) => {
+        let currentSchema = addressSchema;
+        if (isRequired) {
```

Comment:
Perhaps use a ternary operator to reassign the requiredFields (would have to be declared using `let`) variable similar to on line 32?
`requiredFields = (required && currentSchema.required.length === 0) ? requiredFields : [];`
then just 
`currentSchema = _.set('required', requiredFields, currentSchema);`